ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -0,0 +1,134 @@
+package com.microsoft.azure.kusto.ingest;
+
+import com.microsoft.azure.kusto.data.ConnectionStringBuilder;
+import com.microsoft.azure.kusto.data.StreamingClient;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException;
+import com.microsoft.azure.kusto.ingest.result.IngestionResult;
+import com.microsoft.azure.kusto.ingest.source.BlobSourceInfo;
+import com.microsoft.azure.kusto.ingest.source.FileSourceInfo;
+import com.microsoft.azure.kusto.ingest.source.ResultSetSourceInfo;
+import com.microsoft.azure.kusto.ingest.source.StreamSourceInfo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.net.URISyntaxException;
+
+/**
+ * <p>ManagedStreamingIngestClient</p>
+ * <p>
+ * This class combines a managed streaming client with a queued streaming client, to create an optimized experience.
+ * Since the streaming client communicates directly with the engine, it's more prone to failure, so this class
+ * holds both a streaming client and a queued client.
+ * It tries {@value MAX_RETRY_CALLS} times using the streaming client, after which it falls back to the queued streaming client in case of failure.
+ * <p>
+ * Note that {@code ingestFromBlob} acts differently from the other functions - since if you already got a blob it's faster to
```

Comment:
This is not faster - it just makes more sense .. we might want to suggest the user to set the flushImmediately flat to true in the ingestionProperties to get latency similar to streamingIngest

Reply:
FlushImmediately is not recommended for customers (@vladikbr to keep me honest).